src/sage/combinat/permutation.py

```diff
@@ -5382,154 +5386,139 @@ def rewind(L, k):
+        A n-th root of the permutation ``self`` is a permutation `\gamma` such that `\gamma^n == self`.
 
-        Note that the number of k-th roots only depend on the cyclic type of `self`.
+        Note that the number of n-th roots only depend on the cyclic type of ``self``.
 
-        INPUT:
-        
-        - k -- optional integer (default 2), at least 1
-        
         EXAMPLES::
         
             sage: Sigma = Permutations(5).identity()
-            sage: Sigma.number_of_kth_roots(3)
+            sage: Sigma.number_of_nth_roots(3)
             21
             
             sage: Sigma = Permutation('(1, 3)')
-            sage: Sigma.number_of_kth_roots()
+            sage: Sigma.number_of_nth_roots(2)
             0
         
         .. SEEALSO::
         
-                * :meth:`kth_roots`
-                * :meth:`has_kth_root`
+                * :meth:`nth_roots`
+                * :meth:`has_nth_root`
         
         TESTS:
             
         We compute the number of square roots of the identity (i.e. involutions in `S_n`, :oeis:`A000085`), then the number of cubic roots::
             
-            sage: [Permutations(n).identity().number_of_kth_roots() for n in range(2, 10)]
+            sage: [Permutations(n).identity().number_of_nth_roots(2) for n in range(2, 10)]
             [2, 4, 10, 26, 76, 232, 764, 2620]
             
-            sage: [Permutations(n).identity().number_of_kth_roots(3) for n in range(2, 10)]
+            sage: [Permutations(n).identity().number_of_nth_roots(3) for n in range(2, 10)]
             [1, 3, 9, 21, 81, 351, 1233, 5769]
             
-            sage: Permutation('(1)').number_of_kth_roots()
+            sage: Permutation('(1)').number_of_nth_roots(2)
+            1
+
+            sage: Permutation('').number_of_nth_roots(2)
+            1
+
+            sage: Sigma = Permutations(6).random_element()
+            sage: Sigma.number_of_nth_roots(1)
             1
             
-            sage: Permutations(4).identity().number_of_kth_roots(-1)
+            sage: Permutations(4).identity().number_of_nth_roots(-1)
             Traceback (most recent call last):
             ...
-            ValueError: k must be at least 1
+            ValueError: n must be at least 1
         """
         
         from sage.combinat.partition import Partitions
         from sage.combinat.set_partition import SetPartitions
         from sage.arith.misc import divisors, gcd
         from sage.misc.misc_c import prod
 
-        if k < 1:
-            raise ValueError('k must be at least 1')
+        if n < 1:
+            raise ValueError('n must be at least 1')
         
-        P = Permutations(self.size())
+        Cycles = self.cycle_type().to_exp_dict()
+        Result = 1
+        for m, N in Cycles.items():
+            parts = [x for x in divisors(n) if gcd(m*x, n) == x]
+            Result *= sum(SetPartitions(N, pa).cardinality() *
```

Comment:
It might be helpful to explain this computation in a comment.

Reply:
... and not use caml case for variable names, ie `Cycles -> cycles`, `Result -> result`. Caml case is reserved for class names.